### ISSUE TITLE
Definition of timeout/wait option from configuration file

### ISSUE THREAD

**Issue by LucHermitte** [CS-SI/eodag#538]
It would be interesting to be able to configure timeout/wait option for `download*()` methods from the configuration file.

In case it's already possible, I haven't seen it documented on https://eodag.readthedocs.io/en/stable/getting_started_guide/configure.html

**Comment by sbrunato** [comment]
Hello @LucHermitte , no it is not possible for the moment. Thanks for the suggestion!

**Comment by sbrunato** [comment]
In addition of these parameters, a user should also be able to easily tune:
- [ ] `DEFAULT_STREAM_REQUESTS_TIMEOUT` in `download` settings
- [ ] `HTTP_REQ_TIMEOUT` in `search` settings

**Comment by floriandeboissieu** [comment]
@sbrunato is it already possible to change the search timeout, as specified for provider hydroweb in #973, but formatting it for another provider, e.g. theia?

**Comment by sbrunato** [comment]
> @sbrunato is it already possible to change the search timeout, as specified for provider hydroweb in #973, but formatting it for another provider, e.g. theia?

yes it is available since `v2.11.0`, see https://github.com/CS-SI/eodag/pull/841

**Comment by floriandeboissieu** [comment]
Thanks! If I understand well, it can be specified as follows right?
- in config file, example:
```yaml
theia:
    priority: # Lower value means lower priority (Default: 0)
    search:   # Search parameters configuration
        timeout: 10
    ...
```
- with the python api, example:
```python
from eodag import EODataAccessGateway
dag = EODataAccessGateway()
dag.search_all(
    provider="theia",
    productType="S2_MSI_L2A_MAJA",
    start="2016-01-01",
    end="2016-02-01",
    location="T31TCJ",
    timeout=10
)
```
-  with an environment variable:
```shell
export EODAG__THEIA__SEARCH__TIMEOUT=10
```

**Comment by sbrunato** [comment]
> Thanks! If I understand well, it can be specified as follows right?
> * in config file

yes

> * with the python api

no, timeout cannot be specified this way

> * with an environment variable

yes

**Comment by sbrunato** [comment]
> Ok, with the following ?
> 
> ```python
> dag.providers_config["theia"].search.timeout=10
> ```

Yes, this will work. You can also use [update_providers_config](https://eodag.readthedocs.io/en/stable/notebooks/api_user_guide/3_configuration.html#Add-or-update-a-provider)
